src/weapon/w_053.c

```diff
@@ -84,7 +85,186 @@ void EntityWeaponAttack(Entity* self) {
+    } else {
+        clut = 0;
+        v = 0;
+    }
+
+    config = &D_177000_8017AFC4[params];
+
+    if (self->step == 0) {
+        self->facingLeft = PLAYER.facingLeft;
+        self->ext.weapon.unk82 = 20;
+        if (self->facingLeft != 0) {
+            self->ext.weapon.unk82 = -20;
+        }
+        self->ext.weapon.unk80 = -5;
+        if (g_Player.unk0C & 0x20) {
+            self->ext.weapon.unk80 = 11;
+        }
+        self->posX.i.hi += self->ext.weapon.unk82;
+        self->posY.i.hi += self->ext.weapon.unk80;
+
+        self->primIndex = g_api.AllocPrimitives(PRIM_GT4, 1);
+
+        if (self->primIndex != -1) {
+            prim = &g_PrimBuf[self->primIndex];
+
+            if (config->uFlag != 0) {
+                prim->u0 = prim->u2 = 0xC0;
+                prim->u1 = prim->u3 = 0x80;
+            } else {
+                prim->u0 = prim->u2 = 0x80;
+                prim->u1 = prim->u3 = 0xC0;
+            }
+
+            if (config->vFlag != 0) {
+                prim->v1 = v | 0x40;
+                prim->v0 = v | 0x40;
+                prim->v3 = v;
+                prim->v2 = v;
+            } else {
+                prim->v1 = v;
+                prim->v0 = v;
+                prim->v3 = v | 0x40;
+                prim->v2 = v | 0x40;
+            }
+
+            prim->r0 = prim->r1 = prim->r2 = prim->r3 = config->r;
+            prim->g0 = prim->g1 = prim->g2 = prim->g3 = config->g;
+            prim->b0 = prim->b1 = prim->b2 = prim->b3 = config->b;
+
+            prim->tpage = 0x19;
+            prim->priority = PLAYER.zPriority + 2;
+            prim->drawMode = DRAW_UNK_100 | DRAW_TPAGE2 | DRAW_TPAGE |
+                             DRAW_COLORS | DRAW_UNK02 | DRAW_TRANSP;
+            self->flags = FLAG_UNK_08000000 | FLAG_HAS_PRIMS | FLAG_UNK_20000;
+
+            if ((params == 0) || (params == 0xA) || (params == 0x14) ||
+                (params == 0x1E) || (params > 0x27 && params <= 0x31)) {
+                self->ext.weapon.equipId =
+                    self->ext.weapon.parent->ext.weapon.equipId;
+                SetWeaponProperties(self, 0);
+                self->enemyId = self->ext.weapon.parent->enemyId;
+                self->hitboxWidth = 30;
+                self->hitboxHeight = 32;
+            }
+
+            if ((params & 3) == 0) {
+                g_api.PlaySfx(SFX_WEAPON_SWISH_C);
+            }
+            self->rotZ = config->rotZ;
+            self->ext.weapon.lifetime = config->lifetime;
+            self->ext.weapon.unk7E = 0;
+            self->step++;
+        } else {
+            DestroyEntity(self);
+            return;
+        }
+    }
+
+    if ((params == 0) || (params == 0xA) || (params == 0x14) ||
+        (params == 0x1E) || (params == 0x28)) {
+        factory = params + 1;
+        factory = ((g_HandId + 1) << 0xC) + 0x38 +
+                  ((self->ext.weapon.unk7E + (factory)) << 0x10);
+        g_api.CreateEntFactoryFromEntity(self, factory, 0);
```

Comment:
This is an odd way of constructing a factory that doesn't fit with the other weapon functions. The order of operations and operands is significant.